pkg/server/server.go

```diff
@@ -300,5 +310,34 @@ func (s *Server) AddPreShutdownHook(name string, hook genericapiserver.PreShutdo
 
 // NewServer creates a new instance of Server which manages the KCP api-server.
 func NewServer(cfg *Config) *Server {
-	return &Server{cfg: cfg}
+	s := &Server{cfg: cfg}
+	//During the creation of the server, we know that we want to add the namespace controller.
+	s.postStartHooks = append(s.postStartHooks, postStartHookEntry{
+		name: "start-namespace-controller",
+		hook: s.startNamespaceController,
+	})
+	return s
+}
+
+func (s *Server) startNamespaceController(hookContext genericapiserver.PostStartHookContext) error {
+	kubeClient, err := kubernetes.NewForConfig(hookContext.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	metadata, err := metadata.NewForConfig(hookContext.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	versionedInformer := informers.NewSharedInformerFactory(kubeClient, resyncPeriod)
```

Comment:
@davidfestal I am not sure if this is the correct thing to do with the informer. I assume that we will eventually add more controllers that should use the versionedInformer and we may want to manage it at the `server` struct level?